src/overlays/actors/ovl_En_Grasshopper/z_en_grasshopper.c

```diff
@@ -64,92 +138,990 @@ const ActorInit En_Grasshopper_InitVars = {
+            }
+        }
+    }
+
+    if (attackDealsDamage) {
+        Actor_ApplyDamage(&this->actor);
+        if (this->actor.colChkInfo.health > 0) {
+            Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_BATTA_DAMAGE);
+            EnGrasshopper_SetupDamaged(this, play);
+        } else {
+            EnGrasshopper_SetupDead(this, play);
+        }
+    }
+}
+
+void EnGrasshopper_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnGrasshopper* this = THIS;
+
+    SkelAnime_Update(&this->skelAnime);
+    EnGrasshopper_UpdateDamage(this, play);
+    this->actionFunc(this, play);
+
+    Actor_SetFocus(&this->actor, 0.0f);
+    Actor_SetScale(&this->actor, this->dragonflyScale);
+
+    DECR(this->timer);
+    DECR(this->waitTimer);
+    DECR(this->postBankTimer);
+    DECR(this->drawDmgEffTimer);
+
+    Actor_MoveWithGravity(&this->actor);
+    this->actor.world.pos.x += this->damagedVelocity.x;
+    this->actor.world.pos.y += this->damagedVelocity.y;
+    this->actor.world.pos.z += this->damagedVelocity.z;
+    Math_ApproachZeroF(&this->damagedVelocity.x, 1.0f, 2.0f);
+    Math_ApproachZeroF(&this->damagedVelocity.y, 1.0f, 2.0f);
+    Math_ApproachZeroF(&this->damagedVelocity.z, 1.0f, 2.0f);
+    if ((this->action != EN_GRASSHOPPER_ACTION_FALL) && (this->type != EN_GRASSHOPPER_TYPE_WOODFALL)) {
+        Actor_UpdateBgCheckInfo(play, &this->actor, 35.0f, 60.0f, 60.0f, 0x1D);
+    }
+
+    this->actor.shape.rot.z = this->actor.world.rot.z;
+    if (this->type == EN_GRASSHOPPER_TYPE_GROWS_WHEN_SPAWNED) {
+        f32 targetScale = this->type * 0.01f;
+
+        if (targetScale >= 0.018f) {
+            targetScale = 0.018f;
+        }
+
+        Math_ApproachF(&this->dragonflyScale, targetScale, 0.1f, 0.01f);
+    }
+
+    if ((this->action != EN_GRASSHOPPER_ACTION_FALL) && (this->action != EN_GRASSHOPPER_ACTION_APPROACH_PLAYER)) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.world.rot.y, 5, 0x3E8, 5);
+    }
+
+    EnGrasshopper_UpdateEffects(this, play);
+    if ((this->action != EN_GRASSHOPPER_ACTION_DEAD) && (this->action != EN_GRASSHOPPER_ACTION_FALL)) {
+        CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+        if ((this->action == EN_GRASSHOPPER_ACTION_APPROACH_PLAYER) || (this->action == EN_GRASSHOPPER_ACTION_ATTACK)) {
+            CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+        }
+    }
+}
+
+void EnGrasshopper_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnGrasshopper* this = THIS;
+    Vec3f sEffectOffsetFromTailTop = { 500.0f, 0.0f, 0.0f };
+    Vec3f sZeroVec3f = { 0.0f, 0.0f, 0.0f };
+
+    Matrix_Push();
+
+    if ((this->type != EN_GRASSHOPPER_TYPE_WOODFALL_TEMPLE_FINAL_ROOM) &&
+        (this->type != EN_GRASSHOPPER_TYPE_WOODFALL)) {
+        if ((s8)(sLimbIndexToShadowBodyPartsIndex[limbIndex] >= 0)) {
```

Comment:
Yes, otherwise you don't generate these instructions:
![image](https://user-images.githubusercontent.com/12245827/192598885-9a5357dc-1bc6-4d53-907c-dce3ff5a782a.png)